bullet-featherstone/src/SDFFeatures.cc

```diff
@@ -254,23 +260,30 @@ std::optional<Structure> ValidateModel(const ::sdf::Model &_sdfModel)
           }
 
           rootLink = link;
-          continue;
         }
-
-        linkIndex[link] = linkIndex.size();
-        flatLinks.push_back(link);
       }
 
       return true;
     };
 
-  if (!flattenLinks(_sdfModel))
+  if (rootLink == nullptr && !findRootLink(_sdfModel)) {
     return std::nullopt;
```

Comment:
add comment: "No root link was found in this model"

Reply:
added. [e689f8b](https://github.com/gazebosim/gz-physics/pull/562/commits/e689f8b9513c7aa3e94eea73b2979f883699cf3d)